BasementRenovator.py

```diff
@@ -3485,6 +3485,14 @@ def __init__(self, kind=None, entityList=None):
                         self.groups[groupname] = EntityGroupItem(groupname)
 
                     self.groups[groupname].objects.append(entityItem)
+            else:
+                for kind in entity.kinds:
+                    kindgroups = entity.kinds[kind]
+                    for groupname in kindgroups:
+                        if groupname not in self.groups:
+                            self.groups[groupname] = EntityGroupItem(groupname)
+
+                        self.groups[groupname].objects.append(entityItem)
```

Comment:
wasn't around for this PR, but can you refactor this and the above code to remove reuse?